src/key.rs

```diff
@@ -125,9 +127,8 @@ impl SecretKey {
     /// Converts a `SECRET_KEY_SIZE`-byte slice to a secret key
     #[inline]
     pub fn from_slice(data: &[u8])-> Result<SecretKey, Error> {
-        match data.len() {
-            constants::SECRET_KEY_SIZE => {
-                let mut ret = [0; constants::SECRET_KEY_SIZE];
+        match <[u8; constants::SECRET_KEY_SIZE]>::try_from(data) {
```

Comment:
In general `TryFrom` should be used to **implement** fallible conversion, `try_into()` to **call** fallible conversion. (because `TryFrom` implies `TryInto`) I think it's OK when the types are not generic but maybe keeping the good habit is worth changing?

Reply:
IMHO `try_into` hides information into which specific type the conversion happens and makes code less readable / verifiable by human

Reply:
@kixonil I don't think that is generally true. AFAIK it is idiomatic to prefer `TryInto` over `TryFrom` but if you are not doing trait-bounds at all, `TryFrom` might be better because it doesn't need type annotations.

Reply:
This is possible:

```rust
let x: Type = y.try_into()?;
```

But I think `TryFrom` is OK in this case.